### ISSUE TITLE
[BUG] `avoid-non-null-assertion`: no lint for non-null assertion when it's immediately applied to a `Map<String, Object?>` value.

### ISSUE THREAD

**Issue by solid-yuriiprykhodko** [dart-code-checker/dart-code-metrics#1015]
`Map<String, Object?>` is a structure we commonly use to describe JSON objects incoming from external APIs, so this is a problem that emerges when parsing it.

* **Dart code metrics version:** [4.19.1](https://github.com/solid-yuriiprykhodko/subscript-bang-lint-repro/blob/master/pubspec.lock#L80)
* **Dart sdk version:** Dart SDK version: 2.18.1 (stable) (Tue Sep 13 11:42:55 2022 +0200) on "macos_x64"

**Please show your full configuration:**

Repro here: https://github.com/solid-yuriiprykhodko/subscript-bang-lint-repro

<details>
<summary>Configuration</summary>

```yaml
dart_code_metrics:
  rules:
    - avoid-non-null-assertion
```
</details>

**What did you do? Please include the source code example causing the issue.**

```dart
const k = "key";

Map<String, Object?> m = {};

final a = m[k]!; // no lint

final nullableValue = m[k];
final b = nullableValue!; // lint
```

**What did you expect to happen?**
`avoid-non-null-assertion` lint on `final a = m[k]!;` line.

**What actually happened?**
No lint.

**Are you willing to submit a pull request to fix this bug?**
No.

**Comment by incendial** [comment]
@solid-yuriiprykhodko sorry for the long response, but I think that it's nearly impossible to allow `!` on maps only for specific cases, since it's considered as an expected approach https://dart.dev/null-safety/understanding-null-safety#the-map-index-operator-is-nullable

But we can introduce a flag to highlight all the usages of `!` and you will need to ignore all that are irrelevant. Will that work for you?
